### ISSUE TITLE
A few comments on "How to minimize docker image size: 4 key approaches"

### ISSUE THREAD

**Issue by ig0rmin** [hatedabamboo/notes.hatedabamboo.me#2]
>Containers have been a dominant force in the Internet landscape for nearly two decades.

"Two decades" is an exaggeration. Containers became a mainstream (and a hype) tech only after Docker appeared (initial release in 2013, **one** decade ago). There used to be other projects like LXC and Mesos before Docker, but they were much more niche and at least in case with LXC the philosophy was spinning a whole OS inside rather a "container per service". And still even these early containers were released only in 2008 and 2009 correspondingly, about 15 years ago. **Two dacades** ago (2004) there was no most of the tech necessary for building containers (e.g. cgroups) not to say the containers itself. 

Better change it to "nearly one decade".

>RUN /venv/bin/pip install --disable-pip-version-check -r /requirements.txt

Why you need venv in the container? Python virtual environments are useful on the host to have separate dev environments per app, but inside the container we're already in the isolated, visualized environment.

> Level 5: Legendary

I like Go for the ability to deploy it in a scratch container myself, but rewriting the app from Python to Go is unrelated to minimizing the container size. If you started with Python, better stick to Python.

**Comment by hatedabamboo** [comment]
@ig0rmin Hey, thank you for the first ever issue!

> "Two decades" is an exaggeration

It is, in fact, an exaggeration. No more, no less.

> Why you need venv in the container?

By utilizing virtual environment in the setup we're limiting the amount of packages and additional libraries needed for the application only to those necessary for the application itself. Which are contained in the venv. The purpose of using virutal environment is not the isolation of the application environment, but keeping the distributed packages to a minimum.

> If you started with Python, better stick to Python.

During the research for the article I've spent a huge amount of time trying to make a statically compiled binary from the python application. It didn't go well, as I mentioned in the article. Because of that I took a different approach.
